packages/account/src/wallet/base-wallet-unlocked.ts

```diff
@@ -77,12 +77,8 @@ export class BaseWalletUnlocked extends Account {
    * @param transactionRequestLike - The transaction request to sign.
    * @returns A promise that resolves to the signature as a ECDSA 64 bytes string.
    */
-  async signTransaction(transactionRequestLike: TransactionRequestLike): Promise<string> {
-    const transactionRequest = transactionRequestify(transactionRequestLike);
-    const chainId = this.provider.getChain().consensusParameters.chainId.toNumber();
-    const hashedTransaction = transactionRequest.getTransactionId(chainId);
+  async signTransaction(hashedTransaction: string): Promise<string> {
```

Comment:
This method can be made synchronous because the `await` in `const signature = await this.signer().sign(hashedTransaction);` is unnecessary. It's been bugging me for a while but I never really touched this part of the code :D It'd be great if you would change it.

Reply:
On the topic of the changed interface of the method, I don't fully understand the motivation in splitting it up as it seemed like a cohesive piece of code. Could you please explain why you split did it?

Reply:
Couple of reasons. Firstly making it compatible with an `Account` and therefore having the same function signature. Wallet extends account, therefore we could support either of those via `addSigners`. The use case came from dev rel who have been working with connectors, so it felt like a safe assumption that we should definitely support the upper level. 
An account can only sign via a connector. I would expect most, if not all of the logic to live within the implementer of the connector. Account is an abstraction and just acts as a middleman, or `throws` if the connector does not support the function (signing, sending etc.). 
So to make these compatible, one had to change. Either `Account.signTransaction()` takes `TransactionRequestLike` or `BaseWalletUnlocked.signTransaction()` takes a string. I opted for the latter as I believe the role of the wallet or account in this instance is just to sign the data, not convert and sign.

Reply:
@danielbate and I had a chat around this and realized that there are issues in the relationship between `FuelConnector` and `Account`. We did a bit of a dive into it and realized that it's too big of a topic for this PR. The way he did it here is okay in the sense of the current design, but we both agreed that it probably shouldn't be the way forward. I'll try getting more info around this and summarizing the situation in an issue/discussion so that the whole team is on the same page.